noir-projects/aztec-nr/aztec/src/test/helpers/utils.nr

```diff
@@ -78,27 +87,16 @@ impl<let N: u32, let M: u32> Deployer<N, M> {
             self.public_keys_hash
         );
         cheatcodes::advance_blocks_by(1);
-        let original_fn = call_interface.get_original();
-        let original_msg_sender = cheatcodes::get_msg_sender();
-        let original_contract_address = get_contract_address();
-        let original_fn_selector = cheatcodes::get_function_selector();
-        let calldata = call_interface.get_args();
-
-        let fn_selector = FunctionSelector::from_field(PUBLIC_DISPATCH_SELECTOR);
-        let calldata = &[call_interface.get_selector().to_field()].append(calldata);
-        cheatcodes::set_fn_selector(fn_selector);
-        cheatcodes::set_contract_address(instance.to_address());
-        cheatcodes::set_msg_sender(original_contract_address);
-        cheatcodes::set_is_static_call(call_interface.get_is_static());
-        cheatcodes::set_calldata(calldata);
-
-        let _result: T = original_fn(());
-
-        cheatcodes::set_fn_selector(original_fn_selector);
-        cheatcodes::set_contract_address(original_contract_address);
-        cheatcodes::set_msg_sender(original_msg_sender);
-        cheatcodes::set_calldata(calldata);
-        cheatcodes::set_is_static_call(call_interface.get_is_static());
+
+        let mut public_context = PublicContext::new(|| {panic( f"Provide args hash manually")});
```

Comment:
Completely unused in the testing environment because we don't need to access `args_hash`. Whenever we perform a call, that value is populated by a lambda that caches `args_hash` in case it's needed in "userland". In this particular case we just go around it, provide a dummy lambda and if we really need it we can do:

```noir
let mut public_context = PublicContext::new(|| {panic( f"Provide args hash manually")});
public_context.args_hash = Option::some(some_args_hash);
```

Reply:
Why do we not need to access `args_hash`?

Reply:
Because here we are at a "top level call", essentially in a virtual "account contract" we impersonate, and that one doesn't need the args hash. It will be correctly populated in the first public call in the chain